### PR TITLE
External Media: support editor changes in WordPress 6.5

### LINKED ISSUE #36186: WP 6.5 compat: unification of the site and post editors
Epic: #33615 

> In WordPress 6.5, the site editor has been refactored to match the post editor UI and behavior closely. 
-- https://make.wordpress.org/core/2024/03/05/unification-of-the-site-and-post-editors-in-6-5/

- [x] Part of it was already addressed in #34921 and #34970. 
- [ ] Same thing needs to happen for `isInserterOpened` in `projects/plugins/jetpack/extensions/shared/external-media/media-service/index.ts`.

### PR DESCRIPTION
**Pull request by jeherve**
Fixes #36186

## Proposed changes:

Add a new conditional to ensure the External Media feature continues to work with WordPress 6.5.

This should also avoid warnings like this one:

```
select( 'core/edit-post' ).isInserterOpened is deprecated since version 6.5. Please use select( 'core/editor' ).isInserterOpened instead. [index.js:77:9](wp-content/plugins/gutenberg/packages/deprecated/src/index.js)
```

### Other information:

- [ ] Have you written new tests for your changes, if applicable?
- [ ] Have you checked the E2E test CI results, and verified that your changes do not break them?
- [ ] Have you tested your changes on WordPress.com, if applicable (if so, you'll see a generated comment below with a script to run)?

## Jetpack product discussion

* No

## Does this pull request change what data or activity we track or use?
* No

## Testing instructions:

* Install the WordPress Beta Tester plugin
* Go to Posts > Add New
* Add a new image block
* Pick one of the external media sources like Pexels
    * You should be able to pick an image from there.
* Go to Tools > WordPress Beta Tester
* Switch to Bleeding edge
* Go to Dashboard > updates and update to a bleeding edge version of WordPress
* Repeat your external media test to ensure the feature continues to work with WordPress 6.5.
* Ensure you do not see the warning above when opening the browser's JavaScript console.

### CODE REVIEW

File: projects/plugins/jetpack/extensions/shared/external-media/media-service/index.ts [hunk 1]
```diff
@@ -196,6 +196,10 @@ const isMediaSourceConnected = async ( source: MediaSource ) =>
  * @returns {boolean} True if the inserter is opened false otherwise.
  */
 const isInserterOpened = (): boolean =>
+	// @to-do: remove exception when Jetpack requires WordPress 6.5.
+	// eslint-disable-next-line
+	// @ts-ignore
+	select( 'core/editor' )?.isInserterOpened?.() ||
```
Comment (jeherve) [inline projects/plugins/jetpack/extensions/shared/external-media/media-service/index.ts]: This doesn't work, I still get a `Uncaught TypeError: o.select(...).isInserterOpened is not a function`. I don't understand why. What would be the best approach here?

**Comment by jeherve** [comment]
I can't figure out how to get rid of the warning, TypeScript drives me a bit crazy.

@renatoagds Since you worked on #35904, do you think you could take a look? I can't figure out how to get rid of the warning when running WordPress 6.5.

Thank you!

**Comment by renatoagds** [comment]
@jeherve sorry for the long delay here, just pushed an update, can you check if it's working for you?

**Comment by jeherve** [comment]
@renatoagds Thank you! This works well for me!

**Review by monsieur-z — APPROVED** [review]
Works as advertised :shipit:

**Pull request merged**